cpp/bench/ann/src/cuvs/cuvs_cagra_wrapper.h

```diff
@@ -250,13 +250,16 @@ void cuvs_cagra<T, IdxT>::set_search_param(const search_param_base& param,
     RAFT_LOG_DEBUG("moving graph to new memory space: %s", allocator_to_string(graph_mem_).c_str());
     // We create a new graph and copy to it from existing graph
     auto mr = get_mr(graph_mem_);
-    *graph_ = raft::make_device_mdarray<IdxT, int64_t>(
-      handle_, mr, raft::make_extents<int64_t>(index_->graph().extent(0), index_->graph_degree()));
 
-    raft::copy(graph_->data_handle(),
-               index_->graph().data_handle(),
-               index_->graph().size(),
+    // Create a new graph, then copy, and __only then__ replace the shared pointer.
+    // We shouldn't use index_->graph() here, because it may be just a view of graph_.
+    auto new_graph = raft::make_device_mdarray<IdxT, int64_t>(handle_, mr, graph_->extents());
+    raft::copy(new_graph.data_handle(),
+               graph_->data_handle(),
```

Comment:
Shouldn't we use `index_->graph()->data_handle()` here? Whether or not it is a view of `graph_`, it shall always point to a graph once the index is loaded.

When the index is just `load()`-ed, we do not populate `graph_`, therefore we should not use that as as a source.

Reply:
You're right, I've updated these lines to use the index_->graph().